consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validator/ForkingValidatorProvider.java

```diff
@@ -65,8 +65,14 @@ public ForkingValidatorProvider(
   }
 
   @Override
-  public Optional<VoteProvider> getVoteProvider() {
-    return resolveValidatorProvider(blockchain.getChainHeadHeader().getNumber()).getVoteProvider();
+  public Optional<VoteProvider> getVoteProviderAtHead() {
+    return resolveValidatorProvider(blockchain.getChainHeadHeader().getNumber())
+        .getVoteProviderAtHead();
+  }
+
+  @Override
+  public Optional<VoteProvider> getVoteProviderAfterBlock(final BlockHeader header) {
+    return resolveValidatorProvider(header.getNumber() + 1).getVoteProviderAtHead();
```

Comment:
The resolver would handle this by yielding an empty forkSchedule for that block. However, you've highlighted that this would make for a good unit test so I'll add one :)

Reply:
Slight clarification...the whole point is that it's looking ahead by one as this is used as part of creating the next block.